serviceconnector/src/validateLinker/ValidateLinkerStep.ts

```diff
@@ -3,20 +3,34 @@
 *  Licensed under the MIT License. See License.md in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizardExecuteStep, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { AzureWizardExecuteStep, GenericTreeItem, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { ThemeColor, ThemeIcon } from "vscode";
 import { IPickLinkerContext } from "../deleteLinker/IPickLinkerContext";
 import { createLinkerClient } from "../linkerClient";
-import { KnownValidationResultStatus } from "@azure/arm-servicelinker";
 
 export class ValidateLinkerStep extends AzureWizardExecuteStep<IPickLinkerContext> {
     public priority: number = 10;
 
     public async execute(context: IPickLinkerContext): Promise<void> {
         const client = await createLinkerClient(context.credentials);
         const response = await client.linker.beginValidateAndWait(nonNullValue(context.sourceResourceUri), nonNullValue(context.linkerName));
+
+        context.activityChildren = [];
+
         for (const detail of nonNullValue(response.validationDetail)) {
-            if (detail.result === KnownValidationResultStatus.Failure) {
+            if (detail.result === "failure") {
+                context.activityChildren.push(new GenericTreeItem(undefined, {
+                    contextValue: `validateResult-${detail.name}`,
+                    label: nonNullValue(detail.name),
+                    iconPath: new ThemeIcon('error', new ThemeColor('testing.iconFailed'))
+                }));
                 throw new Error(detail.description);
```

Comment:
Although one validation failed, I think it's best to still show all other failed/passed validations as activity children. Maybe throw the error after all the children are added?

Reply:
Actually tested this out and for some reason wen there is an error the validation result only contains the failed check. So moving the error after all the children doesn't change anything.

Reply:
I'm not exactly sure why that happens. But that's fine